xbmc/interfaces/json-rpc/AudioLibrary.cpp

```diff
@@ -908,29 +930,19 @@ JSONRPC_STATUS CAudioLibrary::GetAdditionalAlbumDetails(const CVariant &paramete
     return InternalError;
 
   std::set<std::string> checkProperties;
-  checkProperties.insert("genreid");
+  checkProperties.insert("songgenres");
   std::set<std::string> additionalProperties;
   if (!CheckForAdditionalProperties(parameterObject["properties"], checkProperties, additionalProperties))
     return OK;
 
-  for (int i = 0; i < items.Size(); i++)
+  if (additionalProperties.find("songgenres") != additionalProperties.end())
   {
-    CFileItemPtr item = items[i];
-    if (additionalProperties.find("genreid") != additionalProperties.end())
+    for (int i = 0; i < items.Size(); i++)
```

Comment:
Isn't `CFileItemList` iterable now?

Reply:
Yes it is, but I would prefer to leave this loop in the same style as all the other item loops in this unit. To me that makes it more readable